multinet/downloaders/d3_json.py

```diff
@@ -25,52 +26,60 @@ def download(workspace: str, graph: str) -> Any:
     space = get_workspace_db(workspace)
     if not space.has_graph(graph):
         raise GraphNotFound(workspace, graph)
 
     loaded_graph = space.graph(graph)
 
-    node_tables = loaded_graph.vertex_collections()
-    for node_table in node_tables:
-        table_nodes = loaded_graph.vertex_collection(node_table).all()
-
-        for node in table_nodes:
-            node["id"] = node["_key"]
-            del node["_key"]
-            nodes.append(node)
-
-    pattern = re.compile(r"^([^\d_]\w+)_nodes(/.+)")
-    edge_tables = [edef["edge_collection"] for edef in loaded_graph.edge_definitions()]
-    for edge_table in edge_tables:
-        edges = loaded_graph.edge_collection(edge_table).all()
-
-        for edge in edges:
-            source = edge["_from"]
-            target = edge["_to"]
-            source_match = pattern.search(source)
-            target_match = pattern.search(target)
-
-            if source_match and target_match:
-                source = "".join(source_match.groups())
-                target = "".join(target_match.groups())
-
-            edge["source"] = source
-            edge["target"] = target
-            del edge["_from"]
-            del edge["_to"]
-
-            links.append(edge)
-
-    response = make_response(
-        {
-            "nodes": list(generate_filtered_docs(nodes)),
-            "links": list(generate_filtered_docs(links)),
-        }
-    )
+    def node_generator():
+        node_tables = loaded_graph.vertex_collections()
+        for node_table in node_tables:
+            table_nodes = loaded_graph.vertex_collection(node_table).all()
+
+            comma = ""
+            for node in table_nodes:
+                node["id"] = node["_key"]
+                del node["_key"]
+
+                yield f"{comma}{json.dumps(node)}"
+                comma = comma or ","
+
+    def link_generator():
+        edge_tables = [
+            edef["edge_collection"] for edef in loaded_graph.edge_definitions()
+        ]
+        for edge_table in edge_tables:
+            edges = loaded_graph.edge_collection(edge_table).all()
+
+            comma = ""
+            for edge in edges:
+                source = edge["_from"]
+                target = edge["_to"]
+                source_match = table_nodes_pattern.search(source)
+                target_match = table_nodes_pattern.search(target)
+
+                if source_match and target_match:
+                    source = "".join(source_match.groups())
+                    target = "".join(target_match.groups())
+
+                edge["source"] = source
+                edge["target"] = target
+                del edge["_from"]
+                del edge["_to"]
+
+                yield f"{comma}{json.dumps(edge)}"
+                comma = comma or ","
+
+    def d3_json_generator():
+        yield """{"nodes":["""
+        yield from node_generator()
+        yield """],"links":["""
+        yield from link_generator()
+        yield "]}"
```

Comment:
Whoa this is neat.

We may want a version that doesn't include whitespace (since those bytes will grow linearly with total data size).